src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -536,7 +536,7 @@ public function testParseTimestampAsDateTimeObject($yaml, $year, $month, $day, $
         $expected = new \DateTime($yaml);
         $expected->setTimeZone(new \DateTimeZone('UTC'));
         $expected->setDate($year, $month, $day);
-        $expected->setTime($hour, $minute, $second, 1000000 * ($second - (int) $second));
+        $expected->setTime($hour, $minute, (int) $second, (int) (1000000 * ($second - (int) $second)));
```

Comment:
why do we pass a float in the first place? either we should update the data provider, or we should understand what situation it's supposed to test (same in other places)

Reply:
`$second` is a float of seconds + microseconds. In this line, seconds and microseconds are separated.

I could of course update the data provider to provide seconds and microseconds separately. This would probably improve the readability of the test.

Reply:
let's do this yes!

Reply:
Done!